### PR TITLE
add feature to ignore some namespaces

### PR DESCRIPTION
**Pull request by gminiba**
We needed a feature to omit some namespaces from processing, this could be useful for someone else.

### CODE REVIEW

File: helm/templates/deployment.yaml [hunk 1]
```diff
@@ -50,6 +50,10 @@ spec:
             - "-namespaces"
             - {{ .Values.namespaces | quote }}
             {{- end }}
+            {{- if .Values.namespacesIgnore }}
```
Comment (sstarcher) [inline helm/templates/deployment.yaml:53]: I don't believe this is a valid flag.

Comment (gminiba) [reply helm/templates/deployment.yaml:53]: What's wrong with that? I have added it in Values and main.go.
Actually, these changes are already in our production.

Comment (sstarcher) [reply helm/templates/deployment.yaml:53]: ah, when I reviewed it the other files had not been commited.

File: main.go [hunk 2]
```diff
@@ -263,7 +265,23 @@ func informer() {
...
 			// interface that allows us to get metadata easily
 			mObj := obj.(v1.Object)
-			connect(mObj.GetName())
+			if namespacesIgnore == nil || *namespacesIgnore == "" {
+				connect(mObj.GetName())
+			} else {
+				ignore := false
+				for _, namespace := range strings.Split(*namespacesIgnore, ",") {
+					match, err := regexp.MatchString(namespace, mObj.GetName())
+					if err != nil {
```
Comment (sstarcher) [inline main.go:274]: Would it make sense to compile the regex outside of the for loop here and instead do it at the top.

File: main.go [hunk 3]
```diff
@@ -263,7 +265,23 @@ func informer() {
...
+						log.Infof("Regexp error : %s", err)
+					} else if match {
+						log.Infof("Namespace %s is in ignore list", mObj.GetName())
```
Comment (sstarcher) [inline main.go:277]: Should the connect() move to here and break when it matches?

Comment (sstarcher) [reply main.go:277]: The overarching if could be removed and refactored like.  It should allow for less conditional blocks

shouldConnect = true
if namespacesIgnore == nil... {
  shouldConnect = false
  for ... 
    if match shouldConnect == true && break

if shouldconnect 
  connect(mObj)

**Comment by gminiba** [comment]
please check my changes

File: main.go [hunk 4]
```diff
@@ -263,7 +266,18 @@ func informer() {
...
+			for _, re := range namespacesIgnoreRe {
+				if re.FindString(mObj.GetName()) != "" {
+					shouldConnect = false
+					break
+				}
+			}
+			if shouldConnect {
+				connect(mObj.GetName())
+			} else {
+				log.Infof("Namespace %s is in ignore list", mObj.GetName())
```
Comment (sstarcher) [inline main.go:279]: should this log statement be right before the break and this else can go away?

Comment (gminiba) [reply main.go]: makes sense

**Comment by sstarcher** [comment]
Thanks for the contribution

**Pull request merged**